### PR TITLE
DRILL-7164: KafkaFilterPushdownTest is sometimes failing to pattern m…

### PR DESCRIPTION
**Pull request by sohami**
…atch correctly

### CODE REVIEW

**Comment by sohami** [comment]
@aravi5 / @HanumathRao - Please review this PR

File: exec/java-exec/src/test/java/org/apache/drill/PlanTestBase.java [hunk 1]
```diff
@@ -92,7 +91,20 @@ public static void testPlanMatchingPatterns(String query, String[] expectedPatte
 
   public static void testPlanMatchingPatterns(String query, Pattern[] expectedPatterns, Pattern[] excludedPatterns)
     throws Exception {
-    final String plan = getPlanInString("EXPLAIN PLAN for " + QueryTestUtil.normalizeQuery(query), OPTIQ_FORMAT);
+    testPlanMatchingPatterns(query, OPTIQ_FORMAT, expectedPatterns, excludedPatterns);
+  }
+
+  public static void testPlanMatchingPatterns(String query, String planFormat,
```
Comment (aravi5) [inline exec/java-exec/src/test/java/org/apache/drill/PlanTestBase.java]: Minor comment  here. `planFormat` was not exposed to tests earlier. Should we restrict the format to `OPTIQ_FORMAT` and `JSON_FORMAT`

Comment (sohami) [reply exec/java-exec/src/test/java/org/apache/drill/PlanTestBase.java]: Done

**Comment by aravi5** [comment]
The changes look good! +1

File: contrib/storage-kafka/src/test/java/org/apache/drill/exec/store/kafka/KafkaFilterPushdownTest.java [hunk 2]
```diff
@@ -32,10 +32,8 @@
 @Category({KafkaStorageTest.class, SlowTest.class})
 public class KafkaFilterPushdownTest extends KafkaTestBase {
   private static final int NUM_PARTITIONS = 5;
-  private static final String expectedSubStr = "    \"kafkaScanSpec\" : {\n" +
-                                                   "      \"topicName\" : \"drill-pushdown-topic\"\n" +
-                                                   "    },\n" +
-                                                   "    \"cost\"";
+  private static final String expectedPattern = "kafkaScanSpec.*\\n.*\"topicName\" : \"drill-pushdown-topic\"\\n(" +
```
Comment (HanumathRao) [inline contrib/storage-kafka/src/test/java/org/apache/drill/exec/store/kafka/KafkaFilterPushdownTest.java]: All upper case letters might be good here (EXPECTED_PATTERN) ?

**Comment by HanumathRao** [comment]
Changes look good to me +1.

File: contrib/storage-kafka/src/test/java/org/apache/drill/exec/store/kafka/KafkaFilterPushdownTest.java (hunk 2 above)
Comment (sohami) [reply contrib/storage-kafka/src/test/java/org/apache/drill/exec/store/kafka/KafkaFilterPushdownTest.java]: Changed the name. Also rebased on top of latest master.

**Pull request closed without merging**